src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -136,7 +146,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     } catch (InterruptedException | RejectedExecutionException e) {
       allCards = Collections.emptyList();
     }
-
+    if (!preferredSite.isEmpty()) {
+      List<Card> sortedCards = allCards;
+      for (Card card : sortedCards) {
+        if (card.getSource().toLowerCase().equals(preferredSite)) {
+          int index = allCards.indexOf(card);
```

Comment:
Sounds like you need the index. How about doing a normal for loop (with indices) - that way you don't need to get the index using this O(n) operation?